internal/plugin/dataplane_server.go

```diff
@@ -113,25 +110,26 @@ func (dps *DataplaneServer) run(ctx context.Context) {
 // GET /instances
 func (dps *DataplaneServer) GetInstances(ctx *gin.Context) {
 	slog.Debug("get instances request")
-	instances := dps.instanceService.GetInstances()
-	slog.Debug("got instances", "instances", instances)
 
 	response := []common.Instance{}
 
-	for _, instance := range instances {
+	for _, instance := range dps.instances {
 		response = append(response, common.Instance{
 			InstanceId: &instance.InstanceId,
 			Type:       toPtr(mapTypeEnums(instance.Type.String())),
 			Version:    &instance.Version,
 		})
 	}
 
+	slog.Debug("got instances", "instances", response)
+
 	ctx.JSON(http.StatusOK, response)
 }
 
-// (PUT /instances/{instanceId}/configurations)
+// PUT /instances/{instanceId}/configurations
 func (dps *DataplaneServer) UpdateInstanceConfiguration(ctx *gin.Context, instanceId string) {
-	slog.Debug("update instance configuration request", "instanceId", instanceId)
+	correlationId := uuid.New().String()
```

Comment:
I suppose in the future we'll need to extract the correlation ID from the request context, depending on how we will propagate correlation IDs.